app/services/auth/__test__/oAuth.server.test.ts

```diff
@@ -0,0 +1,112 @@
+import { beforeEach, describe, expect, it, vi } from "vitest";
+
+vi.mock("~/services/auth/session.server");
+vi.mock("~/config/config.server", () => ({
+  serverConfig: () => ({
+    BRAK_IDP_OIDC_CLIENT_ID: "client-id",
+    BRAK_IDP_OIDC_CLIENT_SECRET: "client-secret",
+    BRAK_IDP_OIDC_ISSUER: "https://issuer",
+    BRAK_IDP_OIDC_REDIRECT_URI: "https://redirect",
+  }),
+}));
+
+import { loginAsDeveloper } from "~/mocks/auth/mockAuth.server";
+import {
+  AuthenticationProvider,
+  authenticator,
+  type AuthenticationResponse,
+} from "~/services/auth/oAuth.server";
+import { createUserSession } from "~/services/auth/session.server";
+
+type VerifyArgs = { tokens: { accessToken: () => string }; request: Request };
+type StrategyLike = {
+  verify: (a: VerifyArgs) => Promise<AuthenticationResponse>;
+};
+type AuthWithStrategies = {
+  strategies: Map<string, StrategyLike>;
+};
+
+const mockedCreateUserSession = vi.mocked(createUserSession);
+
+const requestURL = "http://localhost/oauth-test";
+const accessToken = "test-access-token-oauth";
+
+function getBEAStrategy(): StrategyLike {
+  const strategyMap = (authenticator as unknown as AuthWithStrategies)
+    .strategies;
+  const strategy = strategyMap.get(AuthenticationProvider.BEA);
+  if (!strategy) throw new Error("BEA strategy not registered");
```

Comment:
Nit: Using "beA", comes from "besonderes elektronisches Anwaltspostfach"...

```suggestion
  if (!strategy) throw new Error("beA strategy not registered");
```